src/sidebar/helpers/groups.ts

```diff
@@ -50,7 +52,10 @@ export function combineGroups(
 
   // Set flag indicating whether user can leave group.
   for (const group of groups) {
-    group.canLeave = allowLeavingGroups(settings) && group.isMember;
```

Comment:
If my understanding is correct, the reason you can't "leave" the Public group is because the user is not actually a member of it. In that case the `isMember` value here is set incorrectly, so that's what we need to fix.

Reply:
I thought all users were supposed to be members of the world group 🤔

Reply:
I suggest we discuss this with Sean when he is next around and figure out how best to model the behavior in the context of the new group roles and permissions.

Reply:
Makes sense

Reply:
I changed the logic as you suggested, but it sadly has some side effects.

The client treats all groups the user is not a member of as featured groups. The first side effect I can notice is this:

![image](https://github.com/user-attachments/assets/5d95670e-cc9d-49a1-a4ed-e2cc07a89bde)

I'm not asure if this has impact anywhere else.